# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/remoteworkers/RemoteWorkerChild.cpp
# Commit: 7802e5f1bd13
# Full Hash: 7802e5f1bd1395f19f2d78636787c41f64192bd7
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2024-04-19 15:22:44
# Regressor Bug: 1769913
# File Overlap Count: 3
# Description:
#   Bug 1769913 - P3 Remove WorkerRunnable::mWorkerPrivate. r=dom-worker-reviewers,asuth
#   
#   WorkerRunnable no longer keeps a raw pointer(mWorkerPrivate) for the associated WorkerPrivate in this patch.
#   Removing the WorkerRunnable::mWorkerPrivate needs to fix the following problems.
#   
# ==============================================================================

diff -r ffc69aa8f2c3 -r 7802e5f1bd13 dom/workers/remoteworkers/RemoteWorkerChild.cpp
--- a/dom/workers/remoteworkers/RemoteWorkerChild.cpp	Fri Apr 19 09:41:57 2024 +0000
+++ b/dom/workers/remoteworkers/RemoteWorkerChild.cpp	Fri Apr 19 09:41:58 2024 +0000
@@ -111,7 +111,7 @@
   MessagePortIdentifierRunnable(WorkerPrivate* aWorkerPrivate,
                                 RemoteWorkerChild* aActor,
                                 const MessagePortIdentifier& aPortIdentifier)
-      : WorkerThreadRunnable(aWorkerPrivate, "MessagePortIdentifierRunnable"),
+      : WorkerThreadRunnable("MessagePortIdentifierRunnable"),
         mActor(aActor),
         mPortIdentifier(aPortIdentifier) {}
 
@@ -933,7 +933,7 @@
           new MessagePortIdentifierRunnable(
               workerPrivate, aOwner,
               mOp.get_RemoteWorkerPortIdentifierOp().portIdentifier());
-      if (NS_WARN_IF(!r->Dispatch())) {
+      if (NS_WARN_IF(!r->Dispatch(workerPrivate))) {
         aOwner->ErrorPropagationDispatch(NS_ERROR_FAILURE);
       }
     } else if (mOp.type() == RemoteWorkerOp::TRemoteWorkerAddWindowIDOp) {